### ISSUE TITLE
allowableValues not displayed in Swagger UI

### ISSUE THREAD

**Issue by wormuth** [springfox/springfox#2529]
- [x] What version of the library are you using? 2.9.2
- [x] What kind of issue is this? Bug report.

TL;DR: Swagger UI does not display the `allowableValues` set for a param of type list correctly. I suppose this is due to an invalid JSON format generated for the resulting param of type `array` which does not match the format of the specification.

Annotating an `@ApiParam` of type `List<String>` with `allowableValues` set:
`...@ApiParam(value = "...", allowableValues = "value1,value2,value3", allowMultiple = true) @RequestParam(value = "expand", required = false) final List<String> expand,
...
`

Results in following JSON:
```
..."parameters": [{
	"name": "expand",
	"in": "query",
	"description": "...",
	"required": false,
	"type": "array",
	"items": {
		"type": "string"
	},
	"collectionFormat": "multi",
	"allowEmptyValue": false,
	"enum": ["value1", "value2", "value3"]
},
...
```
As of [the spec](https://swagger.io/docs/specification/2-0/describing-parameters/) this is not correct, as far as I can see. It should look like this, rather:
```
..."parameters": [{
	"name": "expand",
	"in": "query",
	"description": "...",
	"required": false,
	"type": "array",
	"items": {
		"type": "string",
                "enum": ["value1", "value2", "value3"]
	},
	"collectionFormat": "multi",
	"allowEmptyValue": false
},
...
```
I tried to figure it out using the Swagger Editor.
The original format is not rendered as desired: 
![nok](https://user-images.githubusercontent.com/2778069/42226764-15851750-7ee0-11e8-88b2-d3e71e524f71.png)

Whereas the format manually fitted to spec does render correctly:
![ok](https://user-images.githubusercontent.com/2778069/42226836-4006acdc-7ee0-11e8-9422-136c9c480d7b.png)

I have not been able to create a running test to demonstrate this for now - hope I could make myself clear, though, and have not missed anything obvious :-)

**Comment by wormuth** [comment]
I wonder if anyone out there has the same problem...

Or putting it the other way: anyone out there with "allowableValues" working as expected and using a Springfox v.2.8+?

Would appreciate any hints or tips on this very much :-/

**Comment by dilipkrish** [comment]
I think that is how it used to work. It appears swagger-ui has fixed the problem. Will need to revisit.

**Comment by watery** [comment]
The same happens for `ParameterBuilder()`.`allowMultiple(true)` used in `globalOperationParameters()`: the resulting field only allows a single parameter in UI.

I'm setting it like this:

	return new Docket(DocumentationType.SWAGGER_2)
		.select()
		// ...
		.build()
		// ...
		.forCodeGeneration(true)
		.globalOperationParameters(
				Arrays.asList(
						// ...
						,
						new ParameterBuilder()
						.name("X-Attribute")
						.allowMultiple(true)
						.allowableValues(new AllowableListValues(Arrays.asList("val", "string"))
						.parameterType("header")
						.required(false)
						.modelRef(new ModelRef("string"))
						.build()
				)
		)
		.directModelSubstitute(GregorianCalendar.class, String.class)
		.useDefaultResponseMessages(false);

**Comment by watery** [comment]
@wormuth Have you had a chance to test your modifications in Docket initialization code like in [my previous comment](https://github.com/springfox/springfox/issues/2529#issuecomment-406293015)?

**Comment by wormuth** [comment]
@watery No, I haven't tried since we do not use this approach.
Why don't you give it a try and document your results here?
